### PR TITLE
[Location sharing] - Navigation to Map view from live location message (PSF-888)

### PR DESCRIPTION
**Pull request by mnaturel**
## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Technical
- [ ] Other :

## Content

Add possibility to navigate to a map screen from a running live location share message.
User pins on map and other UI will be implemented in another pull request.

## Motivation and context

Part of #6012 

## Tests

- Ensure to enable location sharing in Settings -> Preferences
- Ensure to enable developer mode in Settings -> Advanced
- Ensure to enable the feature flaf for live location sharing
- Start a live location share in a room
- Press the running live message
- Check a new screen is opened with a map of the world
- Press back
- Stop the live
- Check pressing on the ended live message does nothing

## Tested devices

- [x] Physical
- [x] Emulator
- OS version(s): Android 11

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/LiveLocationShareMessageItemFactory.kt [hunk 1]
```diff
@@ -75,7 +75,8 @@ class LiveLocationShareMessageItemFactory @Inject constructor(
         val height = dimensionConverter.dpToPx(MessageItemFactory.MESSAGE_LOCATION_ITEM_HEIGHT_IN_DP)
 
         return MessageLiveLocationInactiveItem_()
-                .attributes(attributes)
+                // disable the click on this state item
+                .attributes(attributes.copy(itemClickListener = null))
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/LiveLocationShareMessageItemFactory.kt:79]: Maybe we can avoid to add the clickListener before or add a new boolean to disable instead?

File: vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapViewFragment.kt [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import im.vector.app.core.extensions.addFragment
+import im.vector.app.core.platform.VectorBaseFragment
+import im.vector.app.databinding.FragmentLiveLocationMapBinding
+import im.vector.app.features.location.UrlMapProvider
+import javax.inject.Inject
+
+/**
+ * Screen showing a map with all the current users sharing their live location in room.
+ */
+class LocationLiveMapViewFragment @Inject constructor(
```
Comment (ganfra) [inline vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapViewFragment.kt:37]: Maybe start using @AndroidEntryPoint for fragment with field injection?

**Review by ganfra — APPROVED** [review]
Some small remarks, otherwise LGTM

File: vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/LiveLocationShareMessageItemFactory.kt (hunk 1 above)
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/LiveLocationShareMessageItemFactory.kt:79]: I am not sure I can better and more simple for that. Deciding whether the item should be clickable depends on the status of the live which is computed inside this factory using some fields of the summary data. Unless you have a solution in mind?

File: vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapViewFragment.kt (hunk 2 above)
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapViewFragment.kt:37]: Yes, I will try to make the change.

Comment (mnaturel) [reply vector/src/main/java/im/vector/app/features/location/live/map/LocationLiveMapViewFragment.kt]: See [c46aaa2](https://github.com/vector-im/element-android/pull/6092/commits/c46aaa24a56b0ea531702f3216e3925ce0394ff0)

File: vector/src/main/res/layout/fragment_live_location_map.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (bmarty) [inline vector/src/main/res/layout/fragment_live_location_map.xml:1]: Maybe rename the file to something more generic and reusable like `fragment_simple_container.xml`. I have check and a similar fragment layout does not exist yet.

**Review by bmarty — CHANGES_REQUESTED** [review]

File: vector/src/main/java/im/vector/app/core/di/FragmentModule.kt [hunk 4]
```diff
@@ -993,4 +993,9 @@ interface FragmentModule {
     @IntoMap
     @FragmentKey(LocationPreviewFragment::class)
     fun bindLocationPreviewFragment(fragment: LocationPreviewFragment): Fragment
+
+    /*@Binds
+    @IntoMap
+    @FragmentKey(LocationLiveMapViewFragment::class)
+    fun bindLocationLiveMapViewFragment(fragment: LocationLiveMapViewFragment): Fragment*/
```
[the remark this answers is not in the document]
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/core/di/FragmentModule.kt:1000]: I am wondering if we could have this type of auto verification (no commented code) in Danger?

[the remark this answers is not in the document]
Comment (mnaturel) [reply vector/src/main/java/im/vector/app/core/di/FragmentModule.kt]: Removed in [721d7cb](https://github.com/vector-im/element-android/pull/6092/commits/721d7cb6a0387925568a0acd1955fe0efd46e160)

File: vector/src/main/res/layout/fragment_live_location_map.xml (hunk 3 above)
Comment (mnaturel) [reply vector/src/main/res/layout/fragment_live_location_map.xml]: Done in [7bb73ff](https://github.com/vector-im/element-android/pull/6092/commits/7bb73ffb57e954e3a33520daf6dae0ba9f4c6665)

**Review by bmarty — APPROVED** [review]
Thanks for the update!
As discussed, OK for the click listener management.

**Pull request merged**